torch/_logging/_internal.py

```diff
@@ -0,0 +1,359 @@
+    log_names = settings.split(",")
+
+    def get_name_level_pair(name):
+        clean_name = name.replace(INCR_VERBOSITY_CHAR, "")
+        clean_name = clean_name.replace(DECR_VERBOSITY_CHAR, "")
+        level = None
+        if log_registry.is_log(clean_name):
+            if name[0] == INCR_VERBOSITY_CHAR:
+                level = logging.DEBUG
+            elif name[0] == DECR_VERBOSITY_CHAR:
+                level = logging.ERROR
+            else:
+                level = logging.INFO
+
+        return clean_name, level
+
+    log_state = LogState()
+    for name in log_names:
+        name, level = get_name_level_pair(name)
+        if log_registry.is_log(name):
+            assert level is not None
+            log_qname = log_registry.log_alias_to_log_qname[name]
+            if log_registry.supports_verbosity(name):
+                log_state.enable_log(log_qname, level)
+            else:
+                log_state.enable_log(log_qname, logging.DEBUG)
+        elif log_registry.is_artifact(name):
+            log_state.enable_artifact(name)
+        elif _is_valid_module(name):
+            log_registry.register_log(name, name, True)
+            log_state.enable_log(name, level)
+        else:
+            raise ValueError(
+                f"Invalid log settings: {settings}, must be a comma separated list of log or artifact names."
+            )
+
+    return log_state
+
+def _is_valid_module(qname):
+    try:
+        __import__(qname)
+        return True
+    except ImportError:
+        return False
+
+def _update_log_state_from_env():
+    global log_state
+    log_setting = os.environ.get(LOG_ENV_VAR, None)
+    if log_setting is not None:
+        log_state = _parse_log_settings(log_setting)
+
+
+# setup custom handlers
+# if the log level of a component is set to INFO, setup
+# an additional handler to print those messages, because
+# the debug handler is what handles custom objects like guards,
+# bytecode, etc.
+# if the log level of a component is set to DEBUG, allow all
+# string messages and allowed types (other than those off by default)
+def _setup_handlers(create_handler_fn, log):
+    debug_handler = _tag_handler(create_handler_fn())
+    debug_handler.setFormatter(DEFAULT_FORMATTER)
+    debug_handler.setLevel(logging.DEBUG)
+    log.addHandler(debug_handler)
+
+# mark handlers that we've created
+# so we don't modify user handlers
+def _tag_handler(handler):
+    handler.__torch_log_handler = True
+    return handler
+
+
+def _is_torch_handler(handler):
+    return hasattr(handler, "__torch_log_handler")
+
+
+# clears all torch handlers on specified loggers
+def _clear_handlers(log):
+    to_remove = [handler for handler in log.handlers if _is_torch_handler(handler)]
+    for handler in to_remove:
+        log.removeHandler(handler)
+
+def _reset_logs():
+    for log_name in log_registry.get_log_qnames():
+        log = logging.getLogger(log_name)
+        log.setLevel(logging.WARNING)
```

Comment:
This is wrong, the default level for a logger should be NOTSET, not WARNING.

Reply:
We want to display warnings though right? The way this works is that registering a log (ie, torch._dynamo) indicates this is a top-level log, we will not propagate beyond this (what Vishal's issue was talking about) by default we want to display warnings?

For artifact logs, the default is NOTSET because they should propagate up to one of these "terminal loggers"

Reply:
OK.